### ISSUE TITLE
Floating-point formatting does not show the negative sign

### ISSUE THREAD

**Issue by yangle** [google/jsonnet#799]
```
$ echo '{x: -0.5, y: "%f" % self.x}' | jsonnet -
{
   "x": -0.5,
   "y": "0.500000"
}
$ jsonnet --version
Jsonnet commandline interpreter v0.15.0
```

**Comment by sbarzowski** [comment]
Thanks a lot for reporting! Wow, I don't know how such a basic bug survived all these years. 

This happens only when the number is between in (-1, 0) range.
Here is the reason: https://github.com/google/jsonnet/blob/master/stdlib/std.jsonnet#L538. We expect that `sign(x) * floor(abs(x))` has the same sign as `x`. That's not true for numbers in (-1, 0) where `floor(abs(x)) = 0`.

**Comment by yangle** [comment]
Great. Glad to help!

**Comment by sbarzowski** [comment]
Fixed in #801
